pkg/stats/collector.go

```diff
@@ -179,6 +179,11 @@ func (s *BufferedCollector) Run(ctx context.Context) {
 			s.cache = make(keyIndex)
 			go s.send(metrics) // no need to block on this
 		case <-ctx.Done(): // we're done
+			close(s.writes)
```

Comment:
This means canceling the context passed to `Run()` might lead to a panic if `CollectEvent` is called after. Perhaps we should at least document that somewhere.

Reply:
That's right. I'll remove the channel closing, it does no good other than the possible panic..

Reply:
Yes, suggest we just enable Close method that will stop and wait for the collector to complete its job.

Reply:
Thanks, looked at your PR and used some of it.